### DISCUSSION TITLE
ctrl+plus sends "3;5u", control+minus sends "5;5u"

### DISCUSSION THREAD

**Discussion by JPFrancoia** [ghostty-org/ghostty#3498] [Feature Requests, Ideas]
Hello,

First off, amazing software, thank you very much. I really like how the config is done (this has always been a pain point: having a single place where to configure my terminal, with a simple approach).

I have one problem with the `increase_font_size` and `decrease_font_size` actions.
This is the current mappings I have for these actions (I haven't touched them, these are the defaults):

```bash
ctrl  + equal                          increase_font_size:1
ctrl  + minus                          decrease_font_size:1
```

However they respectively send these codes when I actually press the keys: `3;5u` and `5;5u`, and the font doesn't increase/decrease in size.

I looked around and I seem to be the only one with this issue.

Would you have any idea?

OS: Archlinux
Keyboard layout: French (AZERTY)
Desktop: Cinnamon
I built Ghostty from source

![Image](https://github.com/user-attachments/assets/92557a29-9109-4244-99f0-9c4b4c2a6a9b)

**Comment by rockorager** [comment]
Can you run `cat -v` and press these keys both with and without control?

Even better, if you have kitty installed, while in ghostty run `kitten show-key -m kitty` and do the same (press the keys with and without control)

**Comment by rockorager** [comment]
Also is your physical keyboard azerty? I have a qwerty keyboard but if I set the layout to `fr` I can't reproduce this.

**Comment by JPFrancoia** [comment]
Here you are (`kitten show-key -m kitty`):

```bash
num_lock+ENTER RELEASE
CSI 13 ; 129 : 3 u

num_lock+KP_ADD PRESS +
CSI 57413 ; 129 ; 43 u

num_lock+KP_ADD RELEASE
CSI 57413 ; 129 : 3 u

num_lock+KP_SUBTRACT PRESS -
CSI 57412 ; 129 ; 45 u

num_lock+KP_SUBTRACT RELEASE
CSI 57412 ; 129 : 3 u

ctrl+num_lock+LEFT_CONTROL PRESS
CSI 57442 ; 133 u

ctrl+num_lock+KP_ADD PRESS
CSI 57413 ; 133 u

ctrl+num_lock+KP_ADD RELEASE
CSI 57413 ; 133 : 3 u

ctrl+num_lock+KP_SUBTRACT PRESS
CSI 57412 ; 133 u

ctrl+num_lock+KP_SUBTRACT RELEASE
CSI 57412 ; 133 : 3 u

num_lock+LEFT_CONTROL RELEASE
CSI 57442 ; 129 : 3 u
```

And yes, my keyboard is a physical Azerty keyboard. Logitech MX keys master connected over bluetooth.

Thanks for looking into it.

**Comment by pandvan** [comment]
I'm facing the same issue with Archlinux. My keyboard is a Logitech MX keys, QUERTY, italian layout
The issue appears only if I use plus/minus from numpad.

Ghostty installed from Arch's official repo (now at version 1.0.1)

**Comment by edubxb** [comment]
Hello!

Similar problem here, in my case `ctrl+minus` works, but `ctrl+plus` sends `3;5u`.

That [keybiding works perfectly on Kitty](https://github.com/edubxb/dotfiles/blob/main/home/private_dot_config/kitty/kitty.conf#L59).

Debian GNU/Linux Testing, GNOME 47 (on Wayland), keyboard layout:

![imagen](https://github.com/user-attachments/assets/85fae011-3c3d-4000-9925-68593911cd5e)

My keyboard is QWERTY and [QMK](https://qmk.fm/) based.

Ghostty version (from [deb package](https://github.com/mkasberg/ghostty-ubuntu)):

```
ghostty --version
Ghostty 1.0.1

Version
  - version: 1.0.1
  - channel: stable
Build Config
  - Zig version: 0.13.0
  - build mode : builtin.OptimizeMode.ReleaseFast
  - app runtime: apprt.Runtime.gtk
  - font engine: font.main.Backend.fontconfig_freetype
  - renderer   : renderer.OpenGL
  - libxev     : main.Backend.io_uring
  - GTK version:
    build      : 4.16.3
    runtime    : 4.16.7
  - libadwaita : enabled
    build      : 1.6.0
    runtime    : 1.6.2
  - libX11     : enabled
```

**Reply by willbush** [reply]
Not OP. not sure if I was having a similar issue, but I'm using a keyboard with QMK firmware.

```
$ cat -v
^[[43;5u
```

and this gets `C-+` to work for me:

```
keybind = ctrl+shift+physical:equal=increase_font_size:1
```

by plus I mean `+` KC_PLUS from: https://docs.qmk.fm/keycodes_us_ansi_shifted#keycodes

**Reply by richardnpaul** [reply]
The above `keybind = ctrl+shift+physical:equal=increase_font_size:1`  works for me on Ubuntu 24.04 using the same Ubuntu deb as @edubxb  with UK qwerty layout keyboard (MS Natural 4000).

**Reply by lpanebr** [reply]
I have the exact same issue, except that the alternate binding with `+physical:equal` does not work for me.

```shell
$ cat -v

^[[43;5u # Ctrl plus
^[[61;5u # Ctrl equal
```

**Note:** 
I have two `minus` keys and found that:
- `Ctrl+KC_MINS` (**regular minus**) works as expected.
- `Ctrl+KC_KP_MINUS` (**keypad minus** `^[[45;5u`) fails.

Also on QMK Querty
Also sending the numpad plus `KC_PLUS`
Ubuntu 22.04

**Comment by r2p2** [comment]
`ctrl + plus` and `ctrl + minus` are working fine on my side but `ctrl + shift + something else` causes something like `^[[58;5u`. Other terminals seem work.

**Reply by 00-kat** [reply]
This is Fixterms' fault, see https://github.com/ghostty-org/ghostty/discussions/3151#discussioncomment-11678064.

**Comment by 00-kat** [comment]
My guess is that this is #4007.

**Comment by wexman** [comment]
I also have this problem i think. Ghostty 1.1.2, Linux Mint, Zsh, german qwertz Keyboard. When trying to press ctrl+shift+, to reload the config, it writes ;5u to the screen instead.

**Reply by 00-kat** [reply]
Might be #5369 or #4427 though that's Swedish not German.

**Comment by peterjc** [comment]
This reminds me of https://github.com/ghostty-org/ghostty/issues/3899#issuecomment-2569659569 in discussion of a bug in older versions of Ghostty - only plus/minus on the main keyboard were fixed, not plus/minus on the numberpad (which required adding an additional keybinding):

```
keybind=super+kp_add=increase_font_size:1
```
